### ISSUE TITLE
Undefined symbol at import time when subclassing tzinfo

### ISSUE THREAD

**Issue by pganssle** [PyO3/pyo3#234]
## 🐛 Bug Reports

Currently the following code will build just fine:

```rust
#![feature(specialization)]

use pyo3::prelude::*;
use pyo3::types::PyTzInfo;

#[pyclass(extends=PyTzInfo)]
pub struct TzClass {}
```

But if you build it in a rust extension and try to import the module, you get:

```
E   ImportError: /path/to/so/datetime.cpython-36m-x86_64-linux-gnu.so: \
    undefined symbol: PyDateTime_TZInfoType
```

Not quite sure what's going on here, but I think this *should* work.

### 🌍 Environment

 - Your operating system and version: Arch linux
 - Your python version: 3.7.0
 - How did you install python (e.g. apt or pyenv)? Did you use a virtualenv?: pyenv, virtualenv
 - Your rust version (`rustc --version`): rustc 1.30.0-nightly (33b923fd4 2018-08-18)
 - Are you using the latest pyo3 version? Have you tried using latest master (replace `version = "0.x.y"` with `git = "https://github.com/PyO3/pyo3")?` Occurs when building from source

### 💥 Reproducing

See PR #235.

**Comment by pganssle** [comment]
Same issue is observed for `PyDateTime`, presumably all of the datetime objects.

**Comment by kngwyu** [comment]
I think it's simply because type objects of date type are defined in `PyDataTime_CAPI`(see https://github.com/python/cpython/blob/9df346bf98069a87de14a3c2f69009d800994c63/Include/datetime.h#L150-#L159, though I think you know), but our type objects FFI is defined as https://github.com/PyO3/pyo3/blob/master/src/ffi/datetime.rs#L14.
I think we should remove
```rust
   pub static mut PyDateTime_DateType: PyTypeObject;
   pub static mut PyDateTime_TimeType: PyTypeObject;
...
```
or so and use `PyDateTime_CAPI` for type checking

**Comment by konstin** [comment]
Fixed by #237
